src/context.rs

```diff
@@ -76,6 +77,120 @@ impl Context {
             0
         }
     }
+
+    /// Return the stock string for the [dc_stock::StringId].
+    ///
+    /// If the context callback responds with a string to use, e.g. a
+    /// translation, then this string will be returned.  Otherwise a
+    /// default (English) string is returned.
+    pub fn stock_str(&self, id: dc_stock::StringId) -> String {
+        let ptr = self.call_cb(Event::GET_STRING, id as usize, 0) as *mut libc::c_char;
+        if ptr.is_null() {
+            dc_stock::default_string(id)
+        } else {
+            let ret = to_string(ptr);
+            unsafe { free(ptr as *mut libc::c_void) };
+            ret
+        }
+    }
+
+    /// Return stock string, replacing placeholders with provided string.
+    ///
+    /// This replaces both the *first* `%1$s` **and** `%1$d`
+    /// placeholders with the provided string.
+    pub fn stock_str_repl_string<T: AsRef<str>>(
```

Comment:
please use `string` instead of `str` in methods that return `String`s. Otherwise the rusty reader will assume the return value is `&str`

Reply:
You mean `stock_string_repl_string()` right?  Not the <T: ...> bit?

Reply:
yes